db/deploy/main.py

```diff
@@ -0,0 +1,76 @@
+import json
+import os
+import subprocess
+from typing import Dict, Literal
+
+from fastapi import FastAPI, HTTPException, Query, Request
+from google.cloud import logging, secretmanager
+
+app = FastAPI()
+
+# Setup for Google Cloud clients and logging
+SECRET_CLIENT = secretmanager.SecretManagerServiceClient()
+LOGGING_CLIENT = logging.Client()
+SECRET_PROJECT = 'sample-metadata'
+SECRET_NAME = 'liquibase-schema-updater'
+log_name = 'lb_schema_update_log'
+changelog_file = 'project.xml'
+logger = LOGGING_CLIENT.logger(log_name)
+
+
+def read_db_credentials(env: Literal['prod', 'dev']) -> Dict[Literal['dbname', 'username', 'password', 'host'], str]:
+    """Get database credentials from Secret Manager."""
+    try:
+        secret_path = SECRET_CLIENT.secret_version_path(SECRET_PROJECT, SECRET_NAME, 'latest')
+        response = SECRET_CLIENT.access_secret_version(request={'name': secret_path})
+        return json.loads(response.payload.data.decode('UTF-8'))[env]
+    except Exception as e:  # Broad exception for example; refine as needed
+        text = f'Failed to retrieve or parse secrets: {e}'
+        logger.log_text(text, severity='ERROR')
+        raise HTTPException(status_code=500, detail=text) from e
+
+
+@app.post('/execute-liquibase')
+async def execute_liquibase(request: Request, environment: Literal['prod', 'dev'] = Query(default='dev', regex='^(prod|dev)$')):
+    """Endpoint to remotely trigger Liquibase commands on a GCP VM using XML content."""
+    xml_content = await request.body()
+
+    # Temporary file creation with XML content
+    with open(changelog_file, 'wb') as temp_file:
+        temp_file.write(xml_content)
+        temp_file_path = temp_file.name  # Store file path to use later
```

Comment:
This could cause some weird issues if multiple requests are made (eg: one to each environment). You could use tempfile to generate a temp directory, even chdir to that directory to read the next bit, eg:

```python
with tempfile.TemporaryDirectory() as tempdir:
    # Specify the file path within the temporary directory
    with contextlib.chdir(tempdir):
        with open('project.xml', 'w+) as f:
            f.write(xml_content)

        # do the subprocess.run in this indentation with the right cwd and tempdir
```

Reply:
Good point! Have made that change, thank you for that snippet :)